src/sp/it/pl/util/action/ActionManager.kt

```diff
@@ -57,16 +60,9 @@ object ActionManager {
     @IsConfig(name = "Global shortcuts supported", editable = NONE, info = "Whether global shortcuts are supported on this system")
     val isGlobalShortcutsSupported by c(true)
 
-    @IsConfig(name = "Media shortcuts supported", editable = NONE, info = "Whether media shortcuts are supported on this system")
-    private val isMediaShortcutsSupported by c(true)
-
-    /** @return whether the action listening is running */
-    var isActionListening = false
-        private set
-
     @IsConfig(name = "Global shortcuts enabled", info = "Allows using the shortcuts even if application is not focused.")
-    val globalShortcuts by cv(Os.WINDOWS.isCurrent) {
-        v(it).initSync {
+    val globalShortcutsEnabled by cv(isGlobalShortcutsSupported) {
```

Comment:
We do not forbid anyone from using this feature? What are you referring to?

Reply:
if the 'unsupported' evaluates to false user will not be able to change the value of 'enabled' to true as it will become read only. Its not there for the dev, its there for user as well. You effectively disabled global shortcuts on all Linux systems. 'supported' must default to true and 'enabled' must default to isWindows, to prevent Linux users to run into issues. This is how it's been. It's the only way.

Reply:
I reverted this.

Reply:
No! When it is supported, it will be enabled by default and you can switch it on or off at your liking. If it isn't supported, it will start as off and you can not change it. I do not see your problem with that.

Reply:
It is always supported, but must not be always enabled by default. Your code would enable hotkeys on Linux by default.

Below is the way I intended supported and enabled to work:

The initial value of 'enabled' and 'supported' are unrelated. Application must always check whether both are true in order to use the feature. User may want feature enabled even if it is unsupported (for example if he migrates settings across the system.

The only dependency between 'enabled' and 'supported' is that if supported is false enabled is disabled in ui, i.e. user can not change it from ui, although this is not a necessity. He can also still change it from property file. The disabling feature is done by the line `.readOnlyUnless(isGlobalShortcutsSupported)`

Now, the correct initial values are:
- supported=true, because we support all platforms, despite the bugs
- enabled=true iff on windows to prevent problems on Linux

This makes no sense:
`globalShortcutsEnabled by cv(isGlobalShortcutsSupported)`
Whether an option is enabled is not decided by whether it is supported, but whether it is recommended for user to use it. If platform does not support it, that sux, but it should in my opinion still remain enabled.

The only problem with this is that the `enabled` does not show whether the feature is actually enabled. This i not a problem because if supported is false, enabled will be disabled (greyed out), implicating it has no effect. This is the best solution, because it actually displays 3 states:
- whether feature is supported
- whether it is recommended/desired
- whether it is actually turned on

Reply:
Ah, your issues are gone?

That's great news. So you can use global hotkeys, it works (when no window has focus)? Pls confirm. Like `SHIFT+I`.

I will revert this then and ask you for review.

Reply:
https://github.com/sghpjuikit/player/issues/33

> Now it just works without problems in master. What the heck. I will reopen if it returns.

Yes I have tested global shortcuts. They work without problems. I thought you knew this already, that's why I was so confused by your objections.

Reply:
Done.
I have some more changes to do in the class, but I'll make separate PR for that.

Reply:
No, do not always enable them! They should be enabled by default if they are supported, just as I had written previously! This does not make a difference right now, but makes much more sense semantically.

Reply:
Okay here is the deal: Everything using global shortcuts should only check whether they are enabled, as to expose as little internals as possible, which is what I have implemented. If they were to be not supported, they should thus default to false and not be editable in the interface.

What this enables is that the user could force-enable them, even when they are not officially supported, by editing the value in the config-file, if he is apt enough to deal with eventual problems.

Reply:
I couldn't follow you... :/

Reply:
Nvm, I fixed it.

Read-only configs now never set their value from property file. I can conceive a scenario, where this would be a problem, but for now this is ok.